src/server/v2.0/handler/project.go

```diff
@@ -677,6 +692,10 @@ func (a *projectAPI) getProject(ctx context.Context, projectNameOrID interface{}
 }
 
 func (a *projectAPI) validateProjectReq(ctx context.Context, req *models.ProjectReq) error {
+	if req.Metadata.RetentionID != nil && *req.Metadata.RetentionID != "" {
```

Comment:
Replice with len(req.Metadata.RetentionID)>0

Reply:
These two are logically the same, while the original code 
```
if req.Metadata.RetentionID != nil && *req.Metadata.RetentionID != ""
```
tells the logic more straight forward as the code is read, "the passing in retention_id is not given or given an empty string". But the 
```
len(req.Metadata.RetentionID)>0
```
have an indirect logic that len(nil) = 0.